reasoner/planner/AnswerCountEstimator.java

```diff
@@ -641,10 +641,12 @@ private long estimateInferredAnswerCount(Concludable concludable, Set<Variable>
                         ruleSideIds = new HashSet<>(rule.conclusion().pattern().retrieves());
                     }
 
-                    ruleSideVariables = iterate(ruleSideIds)
-                            .filter(id -> rule.condition().conjunction().pattern().retrieves().contains(id))
-                            .map(id -> rule.condition().conjunction().pattern().variable(id)).toSet();
-                    inferredEstimate += answerCountEstimator.estimateAnswers(rule.condition().conjunction(), ruleSideVariables);
+                    for (Rule.Condition.ConditionBranch conditionBranch: rule.condition().branches()) {
+                        ruleSideVariables = iterate(ruleSideIds)
```

Comment:
what does "ruleSide" mean? i think this name is not clear to me (and in the method above)

Reply:
In the conclusion of the rule being triggered. 
ruleSideIds are the the ids in the concludable mapped to the conclusion.

Reply:
ie "the variables coming from the rule"?
Firstly, why not just call all of these ruleIDs? 
Secondly, do you see why this assignment is demonstrating a confusing naming scenario?

Reply:
Because they're not the IDs of rules.
The idea was this: The unifier has a mapping and a reverse mapping, so it has two sides - The concludable side and the conclusion/rule side.
I do not see why it is confusing.